finagle-netty4-http/src/main/scala/com/twitter/finagle/netty4/http/handler/FixedLengthMessageAggregator.scala

```diff
@@ -5,10 +5,25 @@ import com.twitter.util.StorageUnit
 import io.netty.handler.codec.http._
 
 /**
- * netty can chunk fixed length messages so we dechunk them if the content-length
- * is sufficiently small so that streaming clients and servers can access message
- * contents via `content` and `contentString`. Chunked transfer encoded messages
- * still require accessing `reader`.
+ * Aggregates fixed length http messages and emits [[FullHttpMessage]] downstream.
+ *
+ * Http message is considered fixed length if it has `Content-Length` header and
+ * `Transfer-Encoding` header is not `chunked` or if we can deduce actual content
+ * length to be 0 even if `Content-Length` header is not specified (f.ex. from RFCs).
+ *
+ * Fixed length messages may arrive as a series of smaller pieces (hereinafter referred
+ * to as chunks, but not to be confused with http chunks) from the upstream handlers
```

Comment:
What's the difference between HTTP chunk and chunk? What's so special about HTTP chunk besides being an N bytes?

Reply:
I'm not sure, that exactly "http chunk" phrase is defined anywhere, but at least what I mean by chunk in the context of HTTP chunked encoding is a part of http message body, enclosed in `<content length>\r\n` and `\r\n` as specified by chunked encoding format.

In a request with `Transfer-Encoding: chunked` and body `prefix foobar suffix` these are chunks
```
"7\r\nprefix \r\n"
"7\r\nfoobar \r\n"
"6\r\nsuffix\r\n"
```
In a request without `Transfer-Encoding` header and body `prefix foobar suffix` this is the message body that appeared to be grouped into _N bytes_ groups for some reason (f.ex. these parts been transferred in different TCP packets)
```
"prefix "
"foobar "
"suffix"
```

Although word chunked may mean exactly _N bytes_, I personally find it highly overloaded in the context of `FixedLengthMessageAggregator`. This class deals with requests that have `Transfer-Encoding: chunked` header, and requests that do not have it (logic is different for these cases).

So I tried to make this distinction clear. This wording would have helped me a lot, but maybe I'm overcomplicating this.

IDK, maybe we may try rewording this using terms `chunked` and `chunked-encoded`, where `chunked` means split into _N bytes_ groups, and `chunked-encoded` means having `Transfer-Encoding: chunked` header. So then we'll have _chunks of a chunked message_ and _chunks of a chunked-encoded message_. Not sure, whether it'll help...

Reply:
One interesting thing is, from what I read, Netty never really cares how chunks are formatted on the wire (think HEX prefix w/ a chunk size) and [just shoves whatever is available (think part of an actual chunk) into the pipeline](https://github.com/netty/netty/blob/4.1/codec-http/src/main/java/io/netty/handler/codec/http/HttpObjectDecoder.java#L338-L348).

Reply:
That's an interesting find! I didn't know that. Thanks

Maybe for someone who's more familiar with netty than me, the meaning of _chunk_ is clear and I'm just trying to workaround my ignorance here :smile:

Reply:
As we have little to no control over how Netty decodes HTTP streams, we usually just refer to a chunk (HTTP chunk to be precise) as whatever Netty gives us in `HttpContent`.

Also, I appreciate your patience with this. I'm really excited about this PR!